src/foundry/common/grid/base.d.mts

```diff
@@ -0,0 +1,614 @@
+import type { InexactPartial } from "../../../types/utils.d.mts";
+
+/**
+ * The base grid class.
+ */
+declare abstract class BaseGrid {
+  /** The base grid constructor. */
+  constructor(config: BaseGrid.Configuration);
+
+  size: number;
+
+  sizeX: number;
+
+  sizeY: number;
+
+  distance: number;
+
+  units: string;
+
+  style: string;
+
+  thickness: number;
+
+  color: Color;
+
+  alpha: number;
+
+  /**
+   * The grid type (see {@link CONST.GRID_TYPES}).
+   */
+  type: number;
+
+  get isGridless(): boolean;
+
+  get isSquare(): boolean;
+
+  get isHexagonal(): boolean;
+
+  /**
+   * Calculate the total size of the canvas with padding applied, as well as the top-left coordinates of the inner
+   * rectangle that houses the scene.
+   * @param sceneWidth  - The width of the scene.
+   * @param sceneHeight - The height of the scene.
+   * @param padding     - The percentage of padding.
+   */
+  abstract calculateDimensions(
+    sceneWidth: number,
+    sceneHeight: number,
+    padding: number,
+  ): {
```

Comment:
Should this return be refactored to an external type that's more easily referenceable?